libc/string/string_tokpbrkstr.c

```diff
@@ -0,0 +1,875 @@
+			TEST_ASSERT_EQUAL_HEX64(rest[i - 1], rest[i]);
+		}
+	}
+
+	free(asciiStr);
+#endif
+}
+
+
+/*
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_str)
+{
+}
+
+
+TEST_TEAR_DOWN(string_str)
+{
+}
+
+
+TEST(string_str, basic)
+{
+	const char loremIpsum[BUFFSIZE] = LOREM_IPSUM;
+	char *asciiStr = testdata_createCharStr(ASCII_LENGTH);
+
+	/* Standard use of strstr on arrays */
+	TEST_ASSERT_EQUAL_PTR(loremIpsum, strstr(loremIpsum, "Lorem"));
+	TEST_ASSERT_EQUAL_PTR(&loremIpsum[6], strstr(loremIpsum, "Ipsum"));
+	TEST_ASSERT_EQUAL_PTR(&loremIpsum[12], strstr(loremIpsum, "Dolor"));
+
+	/* Simple usage of strstr with ascii set */
+	TEST_ASSERT_EQUAL_PTR(asciiStr, strstr(asciiStr, asciiStr));
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[ASCII_LENGTH / 4], strstr(asciiStr, &asciiStr[ASCII_LENGTH / 4]));
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[ASCII_LENGTH / 3], strstr(asciiStr, &asciiStr[ASCII_LENGTH / 3]));
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[ASCII_LENGTH / 2], strstr(asciiStr, &asciiStr[ASCII_LENGTH / 2]));
+
+
+	free(asciiStr);
+}
+
+
+TEST(string_str, empty_args)
+{
+	const char empty[] = "",
+			   str[] = "Lorem";
+
+
+	char *asciiStr = testdata_createCharStr(ASCII_LENGTH);
+
+	/* Different scenarios of using empty input or output*/
+	TEST_ASSERT_EQUAL_PTR(NULL, strstr(empty, str));
+	/* Passing an empty string as the second parameter to get first */
+	TEST_ASSERT_EQUAL_PTR(asciiStr, strstr(asciiStr, ""));
+
+	TEST_ASSERT_EQUAL_PTR(str, strstr(str, str));
+
+
+	free(asciiStr);
+}
+
+
+TEST(string_str, out_of_ascii)
+{
+	int i;
+
+	char *extAsciiStr = create_extAscii_set();
+
+	for (i = 1; i < sizeof(extAsciiStr); i++) {
+		TEST_ASSERT_EQUAL_PTR(&extAsciiStr[i], strstr(extAsciiStr, &extAsciiStr[i]));
+	}
+
+	free(extAsciiStr);
+}
+
+
+TEST(string_str, strstr_order)
+{
+	int i;
+	char reversStr[ASCII_LENGTH],
+		*testStr;
+
+	testStr = testdata_createCharStr(ASCII_LENGTH);
+
+	TEST_ASSERT_NOT_NULL(reversStr);
+
+	for (i = 0; i < ASCII_LENGTH - 1; i++) {
+		reversStr[i] = testStr[ASCII_LENGTH - i - 2];
+	}
+	reversStr[ASCII_LENGTH - 1] = 0;
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "cba"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "bac"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "bca"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "aa"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "bb"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "cc"));
+	TEST_ASSERT_EQUAL_STRING("abc", strstr("abc", "ab"));
+	TEST_ASSERT_EQUAL_STRING("bc", strstr("abc", "bc"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "ac"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "aabbcc"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "aaaaaa"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "bbbbbb"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "cccccc"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "ccaabb"));
+	TEST_ASSERT_EQUAL_STRING(NULL, strstr("abc", "bbaacc"));
```

Comment:
Those checks are not about order, I'd suggest adding an another test case for it.